src/lib/y2network/interfaces/fake_interface.rb

```diff
@@ -20,18 +20,20 @@
 require "y2network/interface"
 
 module Y2Network
-  # This class represents an interface which does not exists in the system
-  class FakeInterface < Interface
-    class << self
-      # Build connection
-      #
-      # @todo Would be possible to get the name from the connection?
-      #
-      # @param name [String] Interface name
-      # @param conn [ConnectionConfig] Connection configuration related to the
-      #   network interface
-      def from_connection(name, conn)
-        new(name, type: conn.type)
+  module Interfaces
+    # This class represents an interface which does not exists in the system
+    class FakeInterface < Interface
+      class << self
+        # Build connection
+        #
+        # @todo Would be possible to get the name from the connection?
+        #
+        # @param name [String] Interface name
+        # @param conn [ConnectionConfig] Connection configuration related to the
+        #   network interface
+        def from_connection(name, conn)
```

Comment:
one important thing, is virtual interface build on top of fake interface also fake or not?

Reply:
well ... e.g. bond device will be created by wicked as long as it has at least one "non fake" slave (e.g. eht1 - unconfigured physical device, eth3 - fake interface). But will fail when slave(s) are only fake ones. So, It is somehow in between but I tend to see it as non fake device with kind of invalid configuration when only fake slaves are used.